src/Microsoft.OData.Core/UriParser/Parsers/SelectExpandTermParser.cs

```diff
@@ -109,25 +109,37 @@ private void CheckPathLength(int pathLength)
         /// <returns>A parsed PathSegmentToken representing the next segment in this path.</returns>
         private PathSegmentToken ParseSegment(PathSegmentToken previousSegment, bool allowRef)
         {
-            // TODO $count is defined in specification for expand, it is not supported now. Also note $count is not supported with star as expand option.
-            if (this.lexer.CurrentToken.Text.StartsWith("$", StringComparison.Ordinal) && (!allowRef || this.lexer.CurrentToken.Text != UriQueryConstants.RefSegment))
+            if (this.lexer.CurrentToken.Text.StartsWith("$", StringComparison.Ordinal)
+                && (!allowRef || this.lexer.CurrentToken.Text != UriQueryConstants.RefSegment)
+                && this.lexer.CurrentToken.Text != UriQueryConstants.CountSegment)

```

Comment:
Could you explain the logic here?

Reply:
If Path segment starts with `$`, it should only have `$count` or `$ref` when `allowRef` is `true`
e.g
`$expand=NavProp/$ref`
`$expand=NavProp/$count`